utils/estimated_histogram.hh

```diff
@@ -47,13 +47,359 @@
 #include <chrono>
 #include <seastar/core/metrics_types.hh>
 #include <seastar/core/print.hh>
-
-#include <seastar/core/print.hh>
-
 #include "seastarx.hh"
+#include <seastar/core/bitops.hh>
+#include <limits>
 
 namespace utils {
 
+// Returns largest N such that 2^N <= v.
+// Undefined for v == 0.
+inline constexpr size_t pow2_rank(uint64_t v) {
+    return std::numeric_limits<uint64_t>::digits - 1 - count_leading_zeros(v);
+}
+
+template<uint64_t Min, uint64_t Max, size_t NumBuckets>
+class pseudo_floatingpoint_histogram {
+    std::array<uint64_t, NumBuckets> buckets;
```

Comment:
This needs a #include

Reply:
and coding style: _buckets

Reply:
you're right, moving from struct to class